test/NuGet.Core.Tests/NuGet.Packaging.Test/PackageBuilderTest.cs

```diff
@@ -2626,6 +2626,36 @@ public void PackageBuilderWorksWithFileNameWithoutAnExtension()
             }
         }
 
+        [Theory]
+        [InlineData(".txt", "The 'icon' element 'icon.txt' has an invalid file extension. Valid options are .png, .jpg or .jpeg.")]
+        [InlineData(".jpeg", null)]
+        [InlineData(".jpg", null)]
+        [InlineData(".png", null)]
+        [InlineData(".PnG", null)]
+        [InlineData(".jPG", null)]
+        [InlineData(".jpEG", null)]
+        [InlineData("", "The 'icon' element 'icon' has an invalid file extension. Valid options are .png, .jpg or .jpeg.")]
+        public void Icon_IconInvalidExtension_ThrowsException(string fileExtension, string message)
+        {
+            var testDirBuilder = TestDirectoryBuilder.Create();
+            var nuspecBuilder = NuspecBuilder.Create();
+            var rng = new Random();
+
+            var iconFile = $"icon{fileExtension}";
+
+            nuspecBuilder
+                .WithIcon(iconFile)
+                .WithFile(iconFile);
+
+            testDirBuilder
+                .WithFile(iconFile, rng.Next(1, PackageBuilder.MaxIconFileSize))
+                .WithNuspec(nuspecBuilder);
+
+            TestIconPackaging(
+                testDirBuilder: testDirBuilder,
+                exceptionMessage: message);
```

Comment:
It's not clear to me what this test is asserting 😕 

I think it would be better that the helper method does not catch the exception, and then in these tests you can use
```cs
var exception = Assert.Throws<PackagingException(() => TestIconPackaging(testDirBuilder));
Assert.Equal(message, exception.Message);
```

And the tests that don't expect exceptions just call the helper directly.

Also, the tests that don't throw an exception are not asserting anything, hence if `TestIconPackaging` or `PackageBuilder` did nothing at all, these tests would still pass.

Reply:
I will create two helper methods with proper naming to facilitate code reading.

Reply:
Will follow-up in another issue/PR https://github.com/NuGet/Home/issues/10308